### PR TITLE
moar tests passing

### PR DESCRIPTION
**Pull request by krakjoe**
Down to 6 failures ... out of time for today ...

The way value is dtored/reset when unserialized fails is inconsistent for different serializers, don't know if that is intended.

One of the failures remaining is because a cas_token is set unconditionally, but the test seems to expect it to be null still, seems like an anomaly.

Another is because of making objects null in ctor ... I didn't change the test, but making objects null in ctor is a really quite horrible practice, can we leave it in PHP5 ?

### CODE REVIEW

File: tests/experimental/delete_bykey.phpt [hunk 1]
```diff
@@ -21,7 +21,7 @@ var_dump($m->deleteByKey('keffe', ''));
 echo $m->getResultMessage(), "\n";
 var_dump($m->deleteByKey('', 'keffe'));
 echo $m->getResultMessage(), "\n";
-var_dump($m->deleteByKey('keffe', '���as�� �a�sd�f asdf'));
+var_dump($m->deleteByKey('keffe', '���as�� �a�sd�f asdf')); # no spaces allowed
```
Comment (krakjoe) [inline tests/experimental/delete_bykey.phpt]: I know it seems like I just changed these to pass, but it really looks like this is the expected result. strchr is ran on the keys everywhere to check for spaces ...

File: tests/clone.phpt [hunk 2]
```diff
@@ -12,4 +12,7 @@ $m = clone $m;
 
 echo "GOT HERE";
 --EXPECTF--
-Fatal error: Trying to clone an uncloneable object of class Memcached in %s on line %d
\ No newline at end of file
+Fatal error: Uncaught Error: Trying to clone an uncloneable object of class Memcached in %s:6
```
Comment (krakjoe) [inline tests/clone.phpt]: This is just because errors as exceptions now ... is same ...

File: php_memcached.c [hunk 3]
```diff
@@ -3304,6 +3302,7 @@ zend_bool s_unserialize_value (enum memcached_serializer serializer, int val_typ
 
 			PHP_VAR_UNSERIALIZE_INIT(var_hash);
 			if (!php_var_unserialize(value, (const unsigned char **)&payload_tmp, (const unsigned char *)payload_tmp + payload_len, &var_hash)) {
+				zval_ptr_dtor(value);
```
Comment (krakjoe) [inline php_memcached.c]: other serializers should probably be consistent with this

File: php_memcached.c [hunk 4]
```diff
@@ -356,20 +356,17 @@ char *php_memc_printable_func (zend_fcall_info *fci, zend_fcall_info_cache *fci_
...
+		ZVAL_NULL(&id);
 	}
 
-	/* Call the cb */
-	ZVAL_COPY(&params[0], object);
+	ZVAL_UNDEF(&retval);
 
-	fci->params         = params;
-	fci->param_count    = 2;
+	zend_fcall_info_argn(fci, 2, object, &id);
```
Comment (krakjoe) [inline php_memcached.c]: one of the remaining failures is where params are set for another function call ... I tried to use fcall_info_args but failed, I must misunderstand something about parameters, I'm sure someone who knows could do it ...

File: tests/experimental/cas_invalid_key.phpt [hunk 5]
```diff
@@ -11,11 +11,11 @@ error_reporting(0);
...
-var_dump($m->cas(0, ' �� jas kjjhask d ', true, 10));
+var_dump($m->cas(0, ' �� jas kjjhask d ', true, 10)); # no spaces allowed
 echo $m->getResultMessage(), "\n";
 
 --EXPECTF--
 bool(false)
 A BAD KEY WAS PROVIDED/CHARACTERS OUT OF RANGE
 bool(false)
-%rCLIENT ERROR|NOT FOUND%r
+A BAD KEY WAS PROVIDED/CHARACTERS OUT OF RANGE
```
Comment (krakjoe) [inline tests/experimental/cas_invalid_key.phpt]: this appears to be right, strchr is ran on the key to check for spaces ...

**Comment by mkoppanen** [comment]
Looks like the build fails with /tmp/php-memcached-build/memcached-2.2.0/php_memcached.h:23:33: fatal error: Zend/zend_smart_str.h: No such file or directory

**Comment by krakjoe** [comment]
I was hoping it would be green before merging, but let me get this merged and get the other dependencies tested and then I'll work in php7 branch ...

**Pull request merged**